spec/spec.md

```diff
@@ -0,0 +1,2604 @@
+
+# Circuits and Modules
+
+## Circuits
+
+All FIRRTL circuits consist of a list of modules, each representing a
+hardware block that can be instantiated. The circuit must specify the
+name of the top-level module.
+
+    circuit MyTop :
+       module MyTop :
+          ...
+       module MyModule :
+          ...
+
+## Modules
+
+Each module has a given name, a list of ports, and a statement
+representing the circuit connections within the module. A module port is
+specified by its , which may be input or output, a name, and the data
+type of the port.
+
+The following example declares a module with one input port, one output
+port, and one statement connecting the input port to the output port.
+See section [5.1](#connects)
+for details on the connect statement.
+
+    module MyModule :
+       input foo: UInt
+       output bar: UInt
+       bar <= foo
+
+Note that a module definition does *not* indicate that the module will
+be physically present in the final circuit. Refer to the description of
+the instance statement for details on how to instantiate a module
+(section [5.12](#instances)).
+
+## Externally Defined Modules
+
+Externally defined modules consist of a given name, and a list of ports,
+whose types and names must match its external definition.
+
+    extmodule MyExternalModule :
+       input foo: UInt
+       output bar: UInt
+       output baz: SInt
+
+# Types
+
+Types are used to specify the structure of the data held by each circuit
+component. All types in FIRRTL are either one of the fundamental ground
+types or are built up from aggregating other types.
+
+## Ground Types
+
+There are five ground types in FIRRTL: an unsigned integer type, a
+signed integer type, a fixed-point number type, a clock type, and an
+analog type.
+
+### Integer Types
+
+Both unsigned and signed integer types may optionally be given a known
+positive integer bit width.
+
+    UInt<10>
+    SInt<32>
+
+Alternatively, if the bit width is omitted, it will be automatically
+inferred by FIRRTL's width inferencer, as detailed in section
+[9](#width_inference).
+
+    UInt
+    SInt
+
+### Fixed-Point Number Type
+
+In general, a fixed-point binary number type represents a range of
+values corresponding with the range of some integral type scaled by a
+fixed power of two. In the FIRRTL language, the number represented by a
+signal of fixed-point type may expressed in terms of a base integer
+*value* term and a *binary point*, which represents an inverse power of
+two.
+
+The range of the value term is governed by a *width* an a manner
+analogous to integral types, with the additional restriction that all
+fixed-point number types are inherently signed in FIRRTL. Whenever an
+operation such as a `cat` operates on the "bits" of a fixed-point
+number, it operates on the string of bits that is the signed
+representation of the integer value term. The *width* of a fixed-point
+typed signal is the width of this string of bits.
+
+$$\begin{aligned}
```

Comment:
This doesn't render.

Reply:
Renders on latest. 😉 

<img width="858" alt="Screen Shot 2022-02-16 at 01 32 29" src="https://user-images.githubusercontent.com/1018530/154209282-a063b4fe-a12e-4cc9-bfe3-be82c6c920f2.png">